core/src/main/java/io/questdb/cutlass/pgwire/modern/PGConnectionContextModern.java

```diff
@@ -1298,6 +1319,24 @@ private void shiftReceiveBuffer(long readOffsetBeforeParse) {
         recvBufferReadOffset = 0;
     }
 
+    private void shutdownSocketGracefully() {
+        // calling close on a socket with a
+        // non-empty receive kernel-buffer cause the connection to be RST and
+        // the send buffer discarded and not sent
+
+        socket.shutdown(Net.SHUT_WR); // sends a FIN packet and flushes the send buffer
+        long startTime = System.currentTimeMillis();
```

Comment:
We should be using `MillisecondClock` instead of raw `System` object access

Reply:
fixed